pkg/catalogsourceconfig/update.go

```diff
@@ -0,0 +1,37 @@
+package catalogsourceconfig
+
+import (
+	"context"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	"github.com/sirupsen/logrus"
+)
+
+// NewUpdateReconciler returns a Reconciler that reconciles a
+// CatalogSourceConfig object that needs to be updated
+func NewUpdateReconciler(log *logrus.Entry) Reconciler {
+	return &updateReconciler{
+		log: log,
+	}
+}
+
+// updateReconciler is an implementation of Reconciler interface that
+// reconciles an CatalogSourceConfig object that needs to be updated.
+type updateReconciler struct {
+	log *logrus.Entry
+}
+
+// Reconcile reconciles an CatalogSourceConfig object that needs to be updated.
+// It returns "Configuring" as the next desired phase.
+func (r *updateReconciler) Reconcile(ctx context.Context, in *v1alpha1.CatalogSourceConfig) (out *v1alpha1.CatalogSourceConfig, nextPhase *v1alpha1.Phase, err error) {
+	out = in.DeepCopy()
+
+	// Drop existing Status field so that reconciliation can start anew.
+	out.Status = v1alpha1.CatalogSourceConfigStatus{}
+	nextPhase = phase.GetNext(phase.Configuring)
+
+	r.log.Info("Spec has changed, scheduling for configuring")
+
+	return
```

Comment:
I think it'd be more readable to pass the return variables in the `return` statement rather than name them in the function declaration.

Reply:
This is what we have been doing for other reconcilers, so I am maintaining that pattern here. so this sort of change request is outside the scope of this PR and would need to be requested across the board and we can then revisit it. So please bring this up again especially after we have moved to using Operator SDK v0.1.0.